File: src/main/java/com/netsensia/rivalchess/engine/core/eval/StaticExchangeEvaluator.java
package com.netsensia.rivalchess.engine.core.eval;

import com.netsensia.rivalchess.engine.core.EngineChessBoard;
import com.netsensia.rivalchess.engine.core.type.EngineMove;
import com.netsensia.rivalchess.exception.InvalidMoveException;

public interface StaticExchangeEvaluator {

    int staticExchangeEvaluation(EngineChessBoard board, EngineMove move) throws InvalidMoveException;
}

File: src/main/java/com/netsensia/rivalchess/engine/core/eval/StaticExchangeEvaluator.kt
package com.netsensia.rivalchess.engine.core.eval

import com.netsensia.rivalchess.engine.core.board.EngineBoard
import com.netsensia.rivalchess.engine.core.type.EngineMove
import com.netsensia.rivalchess.exception.InvalidMoveException

interface StaticExchangeEvaluator {
    @Throws(InvalidMoveException::class)
    fun staticExchangeEvaluation(board: EngineBoard, move: EngineMove): Int
}